pages/algorithms/visible.md

```diff
@@ -0,0 +1,8 @@
+---
+title: Visible
+key: visible
+---
+
+Elements or text perceivable through sight.
+
+An element, or text within an element, is considered visible if making the element or text fully transparent would cause a difference in the pixels rendered within any area of the viewport that can be reached via scrolling.
```

Comment:
Not sure about "elements or text". Just sounds odd. Would this work?

[[
Content perceived visually

Content is considered visible, if making it fully transparent would result in a difference in the pixels rendered within any area of the viewport. This includes pixels that are reached by scrolling.
]]

[Content is defined in WCAG](https://www.w3.org/TR/WCAG21/#dfn-content)

Reply:
I agree with @nitedog that "elements or text" sounds odd. "Content" could be ambiguous being aligned with text, images, etc. for some readers, who might leave out widgets like buttons, sliders, etc. But if we link to the WCAG definition then it shouldn't be a problem. However, I don't think that the definition of Visible should be "Content perceived visually" because then you would have to define visually. "Content perceived through sight" sounds good to me.

I've also got some concerns about the definition. If I have content rendered with 99% transparency I cannot perceive it through sight. If I make that content fully transparent there will be a difference in the rendered pixels (and an automated tool will pick it up). But I don't agree the initial 99% transparent content is visible.

Reply:
I do agree with using "content" over "elements or text". As for how to go about handling edge cases such as content that is 99% transparent I think there's more value to be gained from a having a definition that caters to the far majority of use cases. If authors make content 99% transparent and it's flagged by an Auto-WCAG rule because the rule considers the content "visible", I honestly think it's fair to require that authors make sure that the content is 100% transparent if they did not intend for it to be visible. That seems like a win-win in any case.

Otherwise, our definition would likely have to include something about contrast requirements for the pixels that would be rendered and the pixels that are actually rendered, which just seems overly complicated.

Reply:
I hadn't looked at it from that perspective. I agree with you that it's fair to require authors to make content 100% transparent if they want to hide it with transparency. I drop my concern about the definition 👍